pennylane/transforms/get_unitary_matrix.py

```diff
@@ -0,0 +1,73 @@
+# Copyright 2018-2021 Xanadu Quantum Technologies Inc.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+A transform to obtain the matrix representation of a circuit.
+"""
+from functools import wraps
+import numpy as np
+from pennylane.wires import Wires
+import pennylane as qml
+
+
+def get_unitary_matrix(fn, wire_order):
+    """Given a QNode, tape, or quantum function along with a list of the wire order, construct the matrix representation"""
+    n_wires = len(wire_order)
+    wire_order = Wires(wire_order)
+
+    @wraps(fn)
+    def wrapper(*args, **kwargs):
+
+        if isinstance(fn, qml.QNode):
+            # user passed a QNode, get the tape
+            fn.construct(args, kwargs)
+            tape = fn.qtape
+
+        elif isinstance(fn, qml.tape.QuantumTape):
+            # user passed a tape
+            tape = fn
+
+        elif callable(fn):
+            # user passed something that is callable but not a tape or qnode.
+            # we'll assume it is a qfunc!
+            tape = qml.transforms.make_tape(fn)(*args, **kwargs)
```

Comment:
Maybe a good idea to put this in a `try/except` block, in case the function that gets passed is actually not a qfunc.

Reply:
Good idea. But I tested and if a regular function is passed,no exception occurs, it just creates an empty tape. So, should we check for that? Should an empty tape.operations be an error? `get_unitary_matrix` will return the identity, which I suppose is technically correct if there is no quantum operation... So perhaps a warning?

Reply:
Personally I think that if there is no operation it would be better to return an expectation and not the identity. It can be misleading to believe that they are actual operations behind the identity matrix.

Reply:
Yeah I came to the same conclusion after thinking about it. Even if it wouldn't be _wrong_ to return the identity, it's almost surely not what the user intended to do, so it'd be wrong in that sense.